e3nn_jax/_src/s2grid.py

```diff
@@ -934,6 +934,115 @@ def to_s2grid(
+    res_beta: int,
+    res_alpha: int,
+    *,
+    quadrature: str,
+    normalization: str = "integral",
+) -> jnp.ndarray:
+    r"""Sample a signal along `beta` on the sphere given by the m=0 coefficients in the spherical harmonics basis.
+
+    The inverse transformation of :func:`legendre_transform_from_s2grid`
+
+    Args:
+        coeffs (`jnp.ndarray`): coefficient array of shape ``(lmax+1,)``
+        res_beta (int): number of points on the sphere in the :math:`\theta` direction
+        res_alpha (int): number of points on the sphere in the :math:`\phi` direction
+        normalization ({'norm', 'component', 'integral'}): normalization of the basis
+        quadrature (str): "soft" or "gausslegendre"
+
+    Returns:
+        `jnp.ndarray`: signal on the sphere of shape ``(y/beta,)``
+    """
+    lmax = coeffs.shape[0] - 1
+
+    _, _, sh_y, _, _ = _spherical_harmonics_s2grid(
+        lmax, res_beta, res_alpha, quadrature=quadrature, dtype=coeffs.dtype
+    )
+
+    n = _normalization(lmax, normalization, coeffs.dtype, "to_s2")
+    sh_y_m0 = sh_y[:, :, 0]
+    sh_y_qw_m0 = jnp.einsum("bi,i->bi", sh_y_m0, n)
+
+    return jnp.einsum("bi,...i->...b", sh_y_qw_m0, coeffs)
+
+
+def legendre_transform_from_s2grid(
+    x_beta: jnp.ndarray,
+    lmax: int,
+    res_beta: int,
+    res_alpha: int,
+    *,
+    quadrature: str,
+    normalization: str = "integral",
+) -> jnp.ndarray:
+    r"""
+    Transform signal on the sphere, and return the m=0 spherical harmonic components.
+    Args:
+        x_beta (`jnp.ndarray`): signal on the sphere along beta; shape ``(y/beta,)``
+        lmax (int): maximum l of the resulting irreps
+        res_beta (int)
+        res_alpha (int)
+        quadrature (str): "soft" or "gausslegendre"
+        normalization ({'norm', 'component', 'integral'}): normalization of the spherical harmonics basis
+
+    Returns:
+        `jnp.ndarray`: coefficient array of shape ``(..., lmax+1)``
+    """
+    assert res_beta == x_beta.shape[0]
+
+    _, _, sh_y, _, qw = _spherical_harmonics_s2grid(
+        lmax, res_beta, res_alpha, quadrature=quadrature, dtype=x_beta.dtype
+    )
+    n = _normalization(lmax, normalization, x_beta.dtype, "from_s2", lmax)
+
+    sh_y_m0 = sh_y[:, :, 0]
+    sh_y_qw_m0 = jnp.einsum("bi,i,b->bi", sh_y_m0, n, qw)
+
+    x_prime = jnp.einsum("bi,...b->...i", sh_y_qw_m0, x_beta)
+    return x_prime
+
+
+def betas_to_spherical_signal(x_beta, res_beta, res_alpha, *, quadrature):
+    """
+    Convert signals along beta to a SphericalSignal symmetric about alpha.
+
+    Args:
+        x_beta (`jnp.ndarray`): signal on the sphere along beta; shape ``(y/beta,)``
+        res_beta (int)
+        res_alpha (int)
+        quadrature (str): either "soft" or "gausslegendre"
+    Returns:
+        `SphericalSignal`: signal on the sphere of shape ``(y/beta, alpha)``
+    """
+    return SphericalSignal(
+        jnp.repeat(x_beta.reshape((res_beta, 1)), res_alpha, axis=0), quadrature
```

Comment:
Are you sure about the repeat in axis 0?

Can you make it more flexible to the input shape?